pkg/k8s/apis/cilium.io/v2alpha1/lbipam_types.go

```diff
@@ -69,11 +69,20 @@ type CiliumLoadBalancerIPPoolSpec struct {
 	//
 	// +kubebuilder:validation:Optional
 	ServiceSelector *slimv1.LabelSelector `json:"serviceSelector"`
+	// AllowFirstLastIPs, if set to yes means that the first and list IPs of each CIDR will be allocatable.
+	// If no, these IPs will be reserved.
+	//
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:validation:Enum=yes;no
+	AllowFirstLastIPs string `json:"allowFirstLastIPs,omitempty"`
```

Comment:
Because the value is optional, this provides a tri-state type which allows us to preserve existing behavior for users who are upgrading with an existing policy. This could also be done with a `*bool`, however when I brought this up in the community meeting this was preferred, also seems to be more inline with https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#primitive-types

Reply:
@dylandreimerink thanks for the clarification. Note that the API conventions also recommend using string-type aliases instead of enums. For example:

```go
{
...
    AllowFirstLastIPs AllowFirstLastIPType `json:"allowFirstLastIPs,omitempty"`

}

type AllowFirstLastIPType string

const (
    NoAllowFirstLastIPType AllowFirstLastIPType = "No"
    YesAllowFirstLastIPType AllowFirstLastIPType = "Yes"
)
```

Reply:
That is a good point, thanks, I added string-type aliases :+1: